### PR TITLE
[Improve] [Connector-V2] File Connector add lzo compression way.

### PR DESCRIPTION
**Pull request by lightzhao**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java [hunk 1]
```diff
@@ -50,8 +50,12 @@ public class BaseFileSinkConfig implements DelimiterConfig, CompressConfig, Seri
 
     public BaseFileSinkConfig(@NonNull Config config) {
         if (config.hasPath(BaseSinkConfig.COMPRESS_CODEC.key())) {
-            throw new FileConnectorException(CommonErrorCode.UNSUPPORTED_OPERATION,
-                    "Compress not supported by SeaTunnel file connector now");
+            if("lzo".equals(config.getString(BaseSinkConfig.COMPRESS_CODEC.key()))){
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: Please format the code. Thanks.

Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: Format code completed, thank you.

**Comment by TyrantLucifer** [comment]
IMO, using the enum to limit the method of compress is better. Orc and parquet file format also have other compress method

**Comment by lightzhao** [comment]
> IMO, using the enum to limit the method of compress is better. Orc and parquet file format also have other compress method

Add compression enum completed.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java [hunk 2]
```diff
@@ -50,7 +50,7 @@ public class BaseFileSinkConfig implements DelimiterConfig, CompressConfig, Seri
 
     public BaseFileSinkConfig(@NonNull Config config) {
         if (config.hasPath(BaseSinkConfig.COMPRESS_CODEC.key())) {
-            if ("lzo".equals(config.getString(BaseSinkConfig.COMPRESS_CODEC.key()))) {
+            if (CompressFormat.LZO.getCompressCodec().equals(config.getString(BaseSinkConfig.COMPRESS_CODEC.key()))) {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: switch is better

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java [hunk 3]
```diff
@@ -115,7 +116,7 @@ private FSDataOutputStream getOrCreateOutputStream(@NonNull String filePath) {
         FSDataOutputStream fsDataOutputStream = beingWrittenOutputStream.get(filePath);
         if (fsDataOutputStream == null) {
             try {
-                if ("lzo".equals(compressCodec)) {
+                if (CompressFormat.LZO.getCompressCodec().equals(compressCodec)) {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: switch is better

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java (hunk 2 above)
Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: done, thanks.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java (hunk 3 above)
Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/TextWriteStrategy.java]: done, thanks.

**Comment by Hisoka-X** [comment]
image
Please add license header.

**Comment by lightzhao** [comment]
> image Please add license header.

done,thanks.

**Comment by TyrantLucifer** [comment]
image

Please fix CI problem.

**Comment by lightzhao** [comment]
> image
> 
> Please fix CI problem.

done, thanks.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/CompressFormat.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.file.config;
+
+import java.io.Serializable;
+
+public enum CompressFormat implements Serializable {
+
+    LZO("lzo");
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/CompressFormat.java]: How About add default value `NONE`?

Comment (lightzhao) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/CompressFormat.java]: done. good idea.

**Comment by Hisoka-X** [comment]
Please modify doc for this feature

**Comment by lightzhao** [comment]
> Please modify doc for this feature

done, thanks.

**Comment by Hisoka-X** [comment]
> > Please modify doc for this feature
> 
> done, thanks.

Hi, also should add change log in here.
image

**Comment by lightzhao** [comment]
> > > Please modify doc for this feature
> > 
> > 
> > done, thanks.
> 
> Hi, also should add change log in here. image

done, thanks.

**Review by Hisoka-X — DISMISSED** [review]
LGTM

**Comment by TyrantLucifer** [comment]
Waiting CI completed.

**Comment by TyrantLucifer** [comment]
Could you please offer test screenshot images to verify this pull request is work? The following points need to be verified:

1. The file name is correct
2. Data can be read normally in hive or spark

**Comment by lightzhao** [comment]
> Could you please offer test screenshot images to verify this pull request is work? The following points need to be verified:
> 
> 1. The file name is correct
> 2. Data can be read normally in hive or spark

hdfs file：
image
hive query：
image

**Comment by TyrantLucifer** [comment]
> > Could you please offer test screenshot images to verify this pull request is work? The following points need to be verified:
> > 
> > 1. The file name is correct
> > 2. Data can be read normally in hive or spark
> 
> hdfs file： image hive query： image

`*.lzo.txt` might be better? We can use the suffix to Identify the file type quickly, BTW could you please test csv file type?

**Comment by lightzhao** [comment]
> csv file type

The traditional method is to end with a compressed code, but it doesn't use the ending method you mentioned. csv is just a separation symbol problem, which has been tested

**Comment by TyrantLucifer** [comment]
> > csv file type
> 
> The traditional method is to end with a compressed code, but it doesn't use the ending method you mentioned. csv is just a separation symbol problem, which has been tested

cc @EricJoy2048

**Comment by lightzhao** [comment]
@TyrantLucifer plesase review and merge , thanks.

**Comment by EricJoy2048** [comment]
> > > Could you please offer test screenshot images to verify this pull request is work? The following points need to be verified:
> > > 
> > > 1. The file name is correct
> > > 2. Data can be read normally in hive or spark
> > 
> > 
> > hdfs file： image hive query： image
> 
> `*.lzo.txt` might be better? We can use the suffix to Identify the file type quickly, BTW could you please test csv file type?

I agree with you, `*.lzo.txt` is better.

**Comment by lightzhao** [comment]
> @lightzhao

done, complete file name ends with *. lzo.txt.

**Comment by lightzhao** [comment]
@TyrantLucifer @Hisoka-X please review .thanks.

**Review by Hisoka-X — APPROVED** [review]

**Comment by lightzhao** [comment]
@EricJoy2048 plesase review .

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**